packages/labs/context/src/lib/context-root.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {ContextKey} from './context-key.js';
+import {ContextRequest, ContextRequestEvent} from './context-request-event.js';
+import {ContextProviderEvent} from './controllers/context-provider.js';
+
+type UnknownContextKey = ContextKey<unknown, unknown>;
+
+/**
+ * Makes all object properties in a type into WeakRefs
+ */
+type MakeWeak<T> = {[P in keyof T]: T[P] extends object ? WeakRef<T[P]> : T[P]};
```

Comment:
Where does `WeakRefs` type comes from? When using an up to date TypeScript version, along with newest lit versions, including `@lit/reactive-element`, I don't *see* this type.

Reply:
I believe its in the `ESNext` and `ES2021` libs pulled in by setting the `lib` in `tsconfig.json`.

Reply:
Right. Probably our huge repo with many packages at work doesn't use ES2021.